services/s3/src/test/java/software/amazon/awssdk/services/s3/internal/settingproviders/UseArnRegionProviderChainTest.java

```diff
@@ -57,6 +57,20 @@ public void systemPropertiesThrowException_shouldUseConfigFile() {
         assertThat(UseArnRegionProviderChain.create().resolveUseArnRegion()).isEqualTo(Optional.of(Boolean.TRUE));
     }
 
+    @Test
+    void resolveUseArnRegion_systemPropertiesNotSpecified_shouldResolveOncePerCall() {
+        String trueConfigFile = getClass().getResource("ProfileFile_true").getFile();
+        System.setProperty(AWS_CONFIG_FILE.property(), trueConfigFile);
+
+        assertThat(UseArnRegionProviderChain.create().resolveUseArnRegion()).isEqualTo(Optional.of(Boolean.TRUE));
+
+        System.clearProperty(AWS_CONFIG_FILE.property());
+        String falseConfigFile = getClass().getResource("ProfileFile_false").getFile();
+        System.setProperty(AWS_CONFIG_FILE.property(), falseConfigFile);
```

Comment:
minor: can you reset these in an `@After` method instead? Slightly safer to do in case this statement doesn't get reached during the test.

Reply:
Split test into two methods.